R/graph.R

```diff
@@ -53,28 +58,56 @@ graph_to_dsmj <- function(graph, is_sorted=FALSE){
         values[[parameters[[j]]]] <- current_param
       }
     }
-    return(list(src=src_index, dest=dest_index, values=data.frame(values)))
+
+    return(list(src=src_index, dest=dest_index, values=as.data.table(values)))
+  }
+
+  getCellReverse <- function(i) {
+    src <- edgelist_table[["to"]][[i]]
+    dest <- edgelist_table[["from"]][[i]]
+
+    src_index <- variables_indices[src] - 1
+    dest_index <- variables_indices[dest] - 1
+
+    values <- list()
+    # Get all the parameter values for a specific cell
+    for (j in 1: length(parameters)) {
+      current_param <- edgelist_table[i][[parameters[[j]]]]
+      if (current_param > 0){
+        values[[parameters[[j]]]] <- current_param
+      }
+    }
+
+    return(list(src=src_index, dest=dest_index, values=as.data.table(values)))
   }
 
   # Sorted list
   cells <- lapply(cells_indices, getCell)
   cells_df <- data.table::data.table(jsonlite::fromJSON(jsonlite::toJSON(cells, auto_unbox = TRUE)))
 
+  if (is_directed == FALSE){
+    cells_reverse <- lapply(cells_indices, getCellReverse)
+    cells_reverse_df <- data.table::data.table(jsonlite::fromJSON(jsonlite::toJSON(cells_reverse, auto_unbox = TRUE)))
+    cells_df <- rbind(cells_df, cells_reverse_df)
+
+  }
+
   if (is_sorted == TRUE){
     data.table::setorder(cells_df, cols = "src", "dest")
   }
 
   # Create the final json
-  hdsm_json <- list(schemaVersion="1.0", name=paste0("april16-graph", "-hdsm"), variables=variables, cells=cells_df)
+  hdsm_json <- list(schemaVersion="1.0", name=paste0("hdsm"), variables=variables, cells=cells_df)
```

Comment:
this should just be dsm_json (no prefix h). Moreover, you should not try to suffix or format the file name. Assume the user pass as parameter the full path location including file name and extension. 

I am not sure if dV8 cares for this parameter, but if necessary, add one more parameter to graph_to_dsmj and pass the "hdsm" via the transform function for now. Any assumptions should be defer to the transforms.